app/features/history/exercise_name/EditHistoryExerciseActions.js

```diff
@@ -10,6 +13,47 @@ export const dismissExercise = () => {
     }
 };
 
-export const saveExerciseName = (setID, exercise) => {
-    return SetsActionCreators.saveHistorySet(setID, exercise);
+export const cancelExercise = (setID) => (dispatch, getState) => {
+    var state = getState();
+
+    Analytics.setCurrentScreen('history');
+
+    cancelExerciseAnalytics(setID, state);
+
+    dispatch({
+        type: DISMISS_HISTORY_EXERCISE
+    });
+};
+
+export const saveExerciseName = (setID, exercise) => (dispatch, getState) => {
+    var state = getState();
+    
+    saveExerciseAnalytics(setID, exercise, state);
+    
+    dispatch(SetsActionCreators.saveHistorySet(setID, exercise));
 };
+
+const saveExerciseAnalytics = (setID, exercise, state) => {
+    let is_working_set = SetsSelectors.getIsCurrentSet(state, setID);
+    let startDate = DurationsSelectors.getEditHistoryExerciseStart(state);
+    let duration = DurationCalculator.getDurationBetween(startDate, new Date());  
+
+    Analytics.logEventWithAppState('save_exercise_name', {
+        value: duration,
+        duration: duration,
+        is_working_set: is_working_set
+    }, state);
+};
+
+const cancelExerciseAnalytics = (setID, state) => {
```

Comment:
rename to logCancelEditExerciseNameAnalytics

Reply:
done